### PR TITLE
add post about getting clojure ASTs

### PR DESCRIPTION
**Pull request by erizocosmico**

### CODE REVIEW

File: content/post/clojure-ast.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+date: 2017-02-25
+title: "The many roads towards a Clojure AST"
+draft: false
+description: "Extracting the AST of a Clojure program is super easy, you could think. It is a lisp, after all. We will be explaining our journey trying to extract the AST of Clojure programs using the `clojure.tools.analyzer` and how it was not as trivial as we thought."
+categories: ["ast", "clojure", "technical"]
+---
+
+## A little bit of context
+
+At source{d} we are starting a new ambitious project, called [bblfsh](https://github.com/bblfsh), a service to be able to parse and extract ASTs from any programming language. As a part of that project, we were developing the `clojure-driver`, which is in charge of parsing and extracting an AST from Clojure source code files.
```
Comment (Serabe) [inline content/post/clojure-ast.md]: s/a service to be able/a service/

File: content/post/clojure-ast.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+```
+
+**Note:** we are using here `clojure.tools.analyzer.jvm`, which is an implementation of `clojure.tools.analyzer` with specific things for the JVM. There is also `clojure.tools.analyzer.js` for ClojureScript.
+
+There is a very important thing to notice here. We passed `1` as an argument to `analyze`, not a string, not a file. 
+
+That's the first rock in our road, we need to actually pass forms to `analyze`, which means we have to actually read a source file and convert it to a form.
+It's ok, Clojure provides functions to do exactly that.
+
+Our first attempt at do this was very naive, just using `read-string`. Unfortunately, a Clojure program can have more than one form in it and `read-string` only reads one form from the string.
```
Comment (Serabe) [inline content/post/clojure-ast.md]: s/Our first attempt at do/Our first attempt at doing/

File: content/post/clojure-ast.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+     recur                parse-recur
+     fn*                  ana/parse-fn*
+     var                  ana/parse-var
+     #_:else              ana/parse-invoke)
+   form env))
+```
+
+Now, if we glue it all toghether, we will get our AST for all the forms analyzed in isolation. Then, the AST of the program would be the list of the ASTs of all its forms.     
+
+You can take a look at the complete implementation of the bblfsh clojure driver [here](https://github.com/erizocosmico/clojure-driver/tree/feature/clj-parser).
```
Comment (Serabe) [inline content/post/clojure-ast.md]: s/erizocosmico/bblfsh/
Yeah, I know it is still not there. But needs to before releasing the post.

File: content/post/clojure-ast.md (hunk 1 above)
Comment (smola) [inline content/post/clojure-ast.md]: I think the project itself is called `babelfish`, but let's check with @mcuadros.

**Comment by erizocosmico** [comment]
Done @Serabe @smola

File: config.yaml [hunk 4]
```diff
@@ -67,7 +67,7 @@ params:
       social:
         github: ikanor
         twitter: minifranky
-    mvader:
+    erizocosmico:
```
Comment (ajnavarro) [inline config.yaml]: If you change this, what happens with your previous posts?

Comment (erizocosmico) [inline config.yaml]: i changed them as well

Comment (ajnavarro) [inline config.yaml]: Not all of them ;) : https://github.com/src-d/blog/blob/9c13539ca85770f12985f895e15aa19e83d4c7cc/content/post/proteus.md

Rebase and change this too.

**Comment by erizocosmico** [comment]
Done @ajnavarro

File: config.yaml (hunk 4 above)
Comment (erizocosmico) [inline config.yaml]: Oooooooh, right, didn't rebase haha

**Comment by erizocosmico** [comment]
@Serabe ping

**Comment by erizocosmico** [comment]
Do not merge, requires changes after I finish the Clojure driver implementation.

**Comment by vmarkovtsev** [comment]
@erizocosmico So is it finished?

**Comment by erizocosmico** [comment]
@vmarkovtsev nope

**Comment by alcortesm** [comment]
Should I delay my review until it is finished, or you want it now?

**Comment by erizocosmico** [comment]
It should not yet be reviewed
On Wed, 7 Jun 2017 at 15:30, Alberto Cortés <notifications@github.com<mailto:notifications@github.com>> wrote:

Should I delay my review until it is finished, or you want it now?

**Comment by campoy** [comment]
I'm going to schedule this blog post for March 6th. Do you think it's doable?

**Comment by erizocosmico** [comment]
I don’t even remember anymore what this post was about or what was wrong with it and needed to be changed :/
On Sat, 17 Feb 2018 at 00:53, Francesc Campoy <notifications@github.com<mailto:notifications@github.com>> wrote:

I'm going to schedule this blog post for March 6th. Do you think it's doable?

**Comment by campoy** [comment]
I could review it once you're happy with the content.
Or should we just simply drop the topic and close the PR?

**Comment by erizocosmico** [comment]
I think that would be better, yes.
On Wed, 21 Feb 2018 at 20:55, Francesc Campoy <notifications@github.com<mailto:notifications@github.com>> wrote:

I could review it once you're happy with the content.
Or should we just simply drop the topic and close the PR?

**Pull request closed without merging**